### ISSUE TITLE
Typescript generation

### ISSUE THREAD

**Issue by CrosRoad95** [microsoft/ClearScript#438]
Title self explain :) i would like to generate typescript definitions, any suggestions how to do this better? right now i manually specify what classes i would like to use to generate types.

approach i think is the best is to hook clearscript  `addhost*` functions and based on that generate types

**Comment by ClearScriptLib** [comment]
Hi @CrosRoad95,

ClearScript does _not_ generate TypeScript declarations for .NET resources.

Sorry!

**Comment by CrosRoad95** [comment]
yes i already figure it out, just need some tip how  to do this? as i said i want to get all types added to js runtime but from what i see clearscript add more types than i declared

**Comment by ClearScriptLib** [comment]
>just need some tip how to do this?

Sorry, this isn't something we've ever investigated, so we have no tips for you. In theory, since all .NET type information is accessible via reflection, it should be possible to generate TypeScript declarations automatically.

However, capturing the _entirety_ of ClearScript's .NET interop via TypeScript declarations – given all the ways in which you can customize that behavior – would be a large project.

In fact, since you can customize ClearScript's behavior at runtime – e.g., by changing the default script access level or installing a new custom attribute loader – it may _not_ actually be possible to do it all via static TypeScript declarations.

>i want to get all types added to js runtime but from what i see clearscript add more types than i declared

Please say more about this. ClearScript should _not_ be exposing anything without your permission. The only exception is the `EngineInternal` object that's reserved for ClearScript's private use.
